### PR TITLE
New features added

### PR DESCRIPTION
**Pull request by NichitaD**
Added features : 
- Integrated Directions API
-Display routes
-Display trip duration
-Added a new button, for reseting the map, after the route is displayed
-Smoother camera moves
-Changed folders name
-Started integrating Firebase

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (S7012MY) [inline .idea/codeStyles/Project.xml:1]: I'm not sure this should be in GIT

File: app/google-services .json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (S7012MY) [inline app/google-services .json:1]: you have a space in naming

Comment (NichitaD) [reply app/google-services .json:1]: I've seen that there are two files named google-servies.jason. Is it my mistake, or should there be two of them?

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (NichitaD) [reply .idea/codeStyles/Project.xml:1]: I don't really know what this is. Could you please explain it?

**Pull request closed without merging**